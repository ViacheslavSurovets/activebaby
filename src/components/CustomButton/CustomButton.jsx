import React from 'react';

const CustomButton = () => (
  <h1>Custom Button</h1>
);

export default CustomButton;
