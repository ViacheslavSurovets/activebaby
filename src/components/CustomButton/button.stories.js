import React from 'react';

export default { title: 'Button' };

export const CustomButton = (  ...rest ) => {

  return (
    <button { ...rest }>hello</button>
  );
};

