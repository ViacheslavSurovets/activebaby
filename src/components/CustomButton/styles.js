import styled, { css } from 'styled-components';
import { commonFontSize } from '@commonStyles';

const stylesButtonDefault = css`
  ${ commonFontSize };
  outline: none;
  min-width: 10rem;
  height: 5rem;
  font-size: 1.6rem;
`;

const styleButton = css`
  color: ${ ( { theme } ) => theme.customButton.customButtonColor };
  border: 1px solid ${ ( { theme } ) => theme.customButton.customButtonBorderColor };
  background-color: ${ ( { theme } ) => theme.white };
&:hover {
    color: ${ ( { theme } ) => theme.white };
    background: ${ ( { theme } ) => theme.black };
  }
`;

const styleButtonOrangeSoft = css`
  color: ${ ( { theme } ) => theme.white };
  background:${ ( { theme } ) => theme.orangeSoft };
  border: 1px solid transparent;
  &:hover {
    color: ${ ( { theme } ) => theme.orangeSoft };
    background:  ${ ( { theme } ) => theme.white };
    border: 1px solid ${ ( { theme } ) => theme.orangeSoft };
  }
`;

const getStyleButton = ( props ) => {
  if ( props.orangeSoft ) {
    return styleButtonOrangeSoft;
  }
  return styleButton;
};

const styleButtonDisplay = css`
  width: 100%;
`;

const getStyleButtonDisplayBlock = ( props ) => {
  if ( props.block ) {
    return styleButtonDisplay;
  }
  return;
};


export const Button = styled.button`
    ${ stylesButtonDefault };
    ${ getStyleButtonDisplayBlock };
    ${ getStyleButton };
     margin-top: ${props => props.marginTop}rem;
     margin-bottom: ${props => props.marginBottom}rem;
     margin-right: ${props=> props.marginRight}rem;
     margin-left: ${props=> props.marginLeft}rem;
     font-size: ${props => props.fontSize}rem;
`;
