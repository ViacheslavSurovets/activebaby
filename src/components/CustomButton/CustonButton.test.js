import React from 'react';
import { shallow } from 'enzyme';
import CustomButton from './CustomButton';


describe ( 'CustomButton', () => {
  it ( 'CustomButton Snapshot', () => {
    expect ( shallow ( <CustomButton /> ) ).toMatchSnapshot ();
  } );
} );
