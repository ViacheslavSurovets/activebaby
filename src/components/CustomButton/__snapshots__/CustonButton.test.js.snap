// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton CustomButton Snapshot 1`] = `ShallowWrapper {}`;
